src/overlays/actors/ovl_Boss_07/z_boss_07.c

```diff
@@ -242,500 +636,7211 @@ ActorProfile Boss_07_Profile = {
+            if ((player->stateFlags3 & PLAYER_STATE3_1000) && !(player->stateFlags3 & PLAYER_STATE3_80000) &&
+                (this->actor.xzDistToPlayer <= 250.0f)) {
+                Boss07_Wrath_ChooseJump(this, play, false);
+            }
+        }
+    }
+
+    if (this->canEvade) {
+        Boss07_Wrath_JumpAwayFromExplosive(this, play);
+    }
+
+    Boss07_Wrath_CheckBombWhips(this, play);
+
+    if (KREG(88) || this->shouldStartDeath) {
+        KREG(88) = false;
+        this->shouldStartDeath = false;
+        Boss07_Wrath_SetupDeathCutscene(this, play);
+    }
+
+    if (this->maxDecayPixels != 0) {
+        u16* earTex = SEGMENTED_TO_K0(gMajorasWrathEarTex);
+        u16* stripesTex = SEGMENTED_TO_K0(gMajoraStripesTex);
+        u16* mouthTex = SEGMENTED_TO_K0(gMajorasWrathMouthTex);
+        u16* bloodshotEyeTex = SEGMENTED_TO_K0(gMajoraBloodshotEyeTex);
+        u16* eyeTex = SEGMENTED_TO_K0(gMajorasWrathEyeTex);
+        u16* maskTex = SEGMENTED_TO_K0(gMajorasMaskWithNormalEyesTex);
+        u16* veinsTex = SEGMENTED_TO_K0(gMajoraVeinsTex);
+        u16* handTex = SEGMENTED_TO_K0(gMajoraHandTex);
+        u16* bodyTex = SEGMENTED_TO_K0(gMajoraBodyTex);
+
+        for (i = 0; i < this->maxDecayPixels; i++) {
+            s32 rand32x64;
+            s32 rand32x16;
+            s32 rand32x32;
+            s32 rand16x16;
+
+            rand16x16 = Rand_ZeroFloat((16 * 16) - 0.01f);
+            rand32x16 = Rand_ZeroFloat((32 * 16) - 0.01f);
+            rand32x32 = Rand_ZeroFloat((32 * 32) - 0.01f);
+            rand32x64 = Rand_ZeroFloat((32 * 64) - 0.01f);
+
+            earTex[rand16x16] = stripesTex[rand32x16] = mouthTex[rand32x32] = bloodshotEyeTex[rand32x32] =
+                eyeTex[rand32x64] = maskTex[rand32x64] = veinsTex[rand32x64] = handTex[rand32x64] = bodyTex[rand32x64] =
+                    0;
+        }
+    }
+
+    Boss07_UpdateDamageEffects(this, play);
+
+    if ((this->landSfxTimer == 1) || (this->landSfxTimer == 4)) {
+        Actor_PlaySfx(&this->actor, NA_SE_EN_LAST2_WALK2_OLD);
+    }
+
+    if ((player->actor.world.pos.y > 100.0f) && (player->actor.world.pos.z < (KREG(82) + -850.0f))) {
+        player->actor.world.pos.z = KREG(82) + -850.0f;
+    }
+
+    if (player->actor.world.pos.y < -300.0f) {
+        player->actor.world.pos.x = player->actor.world.pos.y = player->actor.world.pos.z = 0.0f;
+    }
+}
+
+/**
+ * The whip physics model works by first assuming each point moves freely, using its current velocity and various forces
+ * to find the next position. Then, starting from the base of the whip, each point is moved away from the previous point
+ * until it is a fixed distance away. The new velocity of each point is found from the scaled difference between its
+ * updated position and previous position. The forces applied are
+ * - drag, which reduces the magnitude of each velocity component by a fixed amount (or to zero if smaller)
```

Comment:
Drag isn't really the right term for this. Drag generally only refers to a resistive force proportional to velocity (e.g. terms like `acceleration += c * velocity / mass` or `acceleration += c * SQ(velocity) / mass` would be deserving of being called drag)
I think it'd be better to just call this "deceleration"